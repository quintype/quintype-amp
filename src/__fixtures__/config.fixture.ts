import { Config, PublisherConfig, AMPConfig, OptsTypes } from "../types/config";

export const publisherConfig: PublisherConfig = {
  "cdn-name": "https://thumbor-stg.assettype.com/",
  "text-direction": "ltr",
  "publisher-name": "newslaundry"
};

export const ampConfig: AMPConfig = {
  menu: {
    enabled: true,
    items: [
      {
        title: "Home",
        url: "https://barandbench.com"
      },
      {
        title: "News",
        url: "https://www.barandbench.com/news"
      },
      {
        title: "Dealstreet",
        url: "https://www.barandbench.com/dealstreet"
      },
      {
        title: "Conferences & Paper presentations",
        url: "https://www.barandbench.com/apprentice-lawyer/call-for-papers"
      },
      {
        title: "Recruitment Tracker",
        url: "https://www.barandbench.com/apprentice-lawyer/recruitment-tracker"
      },
      {
        title: "Apprentice Lawyer",
        url: "https://www.barandbench.com/apprentice-lawyer"
      },
      {
        title: "Viewpoint",
        url: "https://www.barandbench.com/view-point"
      },
      {
        title: "Legal Jobs",
        url: "https://www.barandbench.com/legal-jobs"
      }
    ]
  },
  fonts: {
    primary: {
      url: "Open+Sans:300,400,600,700",
      family: '"Open Sans", sans-serif'
    },
    secondary: {
      url: "PT+Serif:400,400italic,700,700italic",
      family: "sans-serif"
    }
  },
  colors: {
    primary: "#294f32",
    secondary: "#763194",
    "footer-background": "#FDBD10",
    "footer-text-color": "#FDBD10",
    "header-background": "#dd0b4d",
    "section-text-color": "#f4e842"
  },
  "logo-url": "https://thumbor-stg.assettype.com/ace/2018-02/f0ba1430-8c21-49a3-a5cf-50c88ce9ec74/martian3.jpg",
  doubleclick: {
    "top-ad": {
      width: 300,
      height: 250,
      "unit-path": "/35096353/amptesting/formats/sticky"
    },
    "body-ad": {
      width: 300,
      height: 250,
      "unit-path": "/60988533/AMP_Body"
    },
    "bottom-ad": {
      width: 300,
      height: 250,
      "unit-path": "/35096353/amptesting/formats/sticky"
    }
  },
  "google-client-id-api": false,
  "invalid-elements-strategy": "redirect-to-web-version",
  "google-analytics-tracking-id": "UA-ABCDEFG"
};

export const opts: OptsTypes = {
  slots: {
    story: {
      "top-slot": {
        script: `
          <script async custom-element="amp-date-countdown" src="https://cdn.ampproject.org/v0/amp-date-countdown-0.1.js"></script>
          <script async custom-template="amp-mustache" src="https://cdn.ampproject.org/v0/amp-mustache-0.2.js"></script>
        `,
        ampHtml: `<amp-date-countdown height="100" class="ts_wrapper" timestamp-seconds="2147483648"
        layout="fixed-height">
        <template type="amp-mustache">
          <div>This is the top slot</div>
          <div>
            {{d}} days, {{h}} hours, {{m}} minutes and {{s}} seconds until <a class="ts_a" href="https://en.wikipedia.org/wiki/Year_2038_problem">Y2K38</a>.
          </div>
        </template>
      </amp-date-countdown>`,
        styles: `.ts_wrapper {
          background-image: linear-gradient(to left, violet, indigo, blue, green, yellow, orange, red);
          filter: brightness(0.9);
          margin: 15px 0;
          line-height: 50px;
          text-align: center;
          display: block;
          color: white;
        }
        .ts_a {
          color: inherit;
        }`
      },
      "bottom-slot": {
        script: `<script async custom-element="amp-dailymotion" src="https://cdn.ampproject.org/v0/amp-dailymotion-0.1.js"></script>`,
        ampHtml: `<amp-dailymotion data-videoid="x3rdtfy"
          layout="responsive"
          data-ui-highlight="FF4081"
          width="480"
          class="storyTopSlot1"
          height="270">
        </amp-dailymotion>`
      }
    }
  }
};

export const config: Config = {
  publisherConfig,
  ampConfig,
  opts
};
