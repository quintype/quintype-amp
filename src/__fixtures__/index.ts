import { config, publisherConfig, ampConfig, configOpts } from "./config.fixture";
import { textStory } from "./text-story.fixture";
import { storyWithManyJsEmbeds } from "./story-with-jsembeds.fixture";
import { relatedStories } from "./related-stories";
import { seo } from "./seo.fixture";
import { allElementsStory } from "./all-element-story.fixture";
import { liveBlogStory } from "./live-blog-story.fixture";

export {
  allElementsStory,
  config,
  publisherConfig,
  ampConfig,
  textStory,
  storyWithManyJsEmbeds,
  relatedStories,
  configOpts,
  seo,
  liveBlogStory
};
