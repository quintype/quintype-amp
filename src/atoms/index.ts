import Carousel from "./carousel";
import { Image } from "./image";
import Layout from "./layout";
import Facebook from "./facebook";
import Twitter from "./twitter";
import { Footer } from "./footer";
import { Section } from "./section";
import { Spacer } from "./spacer";
import { Author } from "./author";
import { DateTime } from "./date-time";
import { StoryElement } from "./story-elements/story-element";
import { StoryElements } from "./story-elements";

export {
  Carousel,
  Image,
  Layout,
  Facebook,
  Twitter,
  Footer,
  Section,
  Spacer,
  Author,
  DateTime,
  StoryElement,
  StoryElements
};
