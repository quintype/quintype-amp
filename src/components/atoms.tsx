import React, { Fragment } from 'react'
import { Helmet } from 'react-helmet'
import * as atomTypes from '../type-declarations/atomicComponents'

export const Image = (props: atomTypes.Image) => {
  return <amp-img {...props}></amp-img>
}

export const Title = (props: atomTypes.Title) => {
  return (
    <Helmet>
      <title>{props.children}</title>
    </Helmet>
  )
}

export const Carousel = (props: atomTypes.Carousel) => {
  return (
    <Fragment>
      <Helmet>
        <script async={undefined} custom-element="amp-carousel" src="https://cdn.ampproject.org/v0/amp-carousel-0.2.js"></script>
        <meta name="lorem ipsum" content="carousel script added" />
      </Helmet>
      <amp-carousel {...props}>{props.children}</amp-carousel>
    </Fragment>
  )
}

export const FitText = (props: atomTypes.FitText) => {
  return (
    <Fragment>
      <Helmet>
        <script async={undefined} custom-element="amp-fit-text" src="https://cdn.ampproject.org/v0/amp-fit-text-0.1.js"></script>
        <meta name="lorem ipsum" content="Fit Text script added" />
      </Helmet>
      <amp-fit-text {...props}>{props.children}</amp-fit-text>
    </Fragment>
  )
}

export const Youtube = (props: atomTypes.Youtube) => {
  return (
    <Fragment>
      <Helmet>
        <script async={undefined} custom-element="amp-youtube" src="https://cdn.ampproject.org/v0/amp-youtube-0.1.js"></script>
        <meta name="lorem ipsum" content="YOUTUBE script added" />
      </Helmet>
      <amp-youtube {...props}>{props.children}</amp-youtube>
    </Fragment>
  )
}

// const Amp = {Image, Title, Carousel, FitText, Youtube}
// export default Amp