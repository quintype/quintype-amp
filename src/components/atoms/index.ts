import Carousel from './Carousel'
import FitText from './FitText'
import Image from './Image'
import Title from './Title'
import Youtube from './Youtube'

export {Carousel,FitText,Image,Title,Youtube}
