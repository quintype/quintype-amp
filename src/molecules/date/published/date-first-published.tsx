import React from "react";
import { withStoryAndConfig } from "../../../context";
import { DatePublishedTypes } from "./types";
import { DatePublishedBase } from "./date-published-base";

export const DateFirstPublishedBase = ({ story, format, prepend, config }: DatePublishedTypes) => {
  return <DatePublishedBase story={story} format={format} prepend={prepend} config={config} type="first" />;
};

/**
 * DateFirstPublished Component - uses library "date-fns-tz" to format the story publish date. Uses the "DateTime" atomic component internally
 *
 * ```js
 * <DateFirstPublished format="pp 'on' do MM yyyy" prepend="Date of publication: " />
 * ```
 *
 * @param {Object} props Object containing parameters passed to the render prop
 * @param {String|Component} props.prepend Optional. String or Component (logo) that is prepended
 * @param {String} props.format Optional. String that is used to format date. The default is "do MMM, yyyy 'at' p". For formatting info, check [the date-fns Docs]{@link https://date-fns.org/docs/format}
 *
 *  @category Molecules
 * @component
 */

export const DateFirstPublished = withStoryAndConfig(DateFirstPublishedBase);
