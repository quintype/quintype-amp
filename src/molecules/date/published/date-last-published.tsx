import React from "react";
import { DateTime } from "../../../atoms";
import { withStoryAndConfig } from "../../../context";
import { DateLastPublishedTypes } from "./types";
import { getHumanizedDateTime } from "../../../utils/date-time";

export const DateLastPublishedBase = ({ story, format, prepend }: DateLastPublishedTypes) => {
  const dateTime = story["last-published-at"];
  if (!dateTime) return null;

  const formatDateTime = format || "do MMM, yyyy 'at' p";
  const humanizedDate = getHumanizedDateTime({
    dateFormat: formatDateTime,
    timeZone: "Asia/Kolkata",
    timeStamp: dateTime
  });

  return <DateTime formattedDate={humanizedDate} prepend={prepend} />;
};

/**
 * DateLastPublished Component - uses library "date-fns-tz" to format the story publish date. Uses the "DateTime" atomic component internally
 *
 * ```js
 * <DateLastPublished format="pp 'on' do MM yyyy" prepend="Date of publication: " />
 * ```
 *
 * @param {Object} props Object containing parameters passed to the render prop
 * @param {String|Component} props.prepend Optional. String or Component (logo) that is prepended
 * @param {String} props.format Optional. String that is used to format date. The default is "do MMM, yyyy 'at' p". For formatting info, check [the date-fns Docs]{@link https://date-fns.org/docs/format}
 *
 *  @category Molecules
 * @component
 */

export const DateLastPublished = withStoryAndConfig(DateLastPublishedBase);
