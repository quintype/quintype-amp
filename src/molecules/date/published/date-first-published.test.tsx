import React from "react";
import { DateFirstPublishedBase } from "./date-first-published";
import { shallow, mount } from "enzyme";
import { DateTime } from "../../../atoms";
import { config } from "../../../__fixtures__";
import { ThemeProvider } from "styled-components";
import defaultTokens from "../../../context/theme/tokens";
import { getDummyStory } from "./date-last-published.test";

describe("DateFirstPublished", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<DateFirstPublishedBase story={getDummyStory()} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should pass date in default format to <DateTime />", () => {
    const wrapper = mount(
      <ThemeProvider theme={defaultTokens}>
        <DateFirstPublishedBase story={getDummyStory()} />
      </ThemeProvider>
    );
    expect(wrapper.find(DateTime).prop("formattedDate")).toBe("12th Feb, 2020 at 4:07 PM");
  });
  it("should pass date in custom format to <DateTime />", () => {
    const wrapper = mount(
      <ThemeProvider theme={defaultTokens}>
        <DateFirstPublishedBase story={getDummyStory()} format="dd MM yyyy" />
      </ThemeProvider>
    );
    expect(wrapper.find(DateTime).prop("formattedDate")).toBe("12 02 2020");
  });
  it("should pass prepend to <DateTime />", () => {
    const wrapper = mount(
      <ThemeProvider theme={defaultTokens}>
        <DateFirstPublishedBase story={getDummyStory()} prepend="lorem ipsum" />
      </ThemeProvider>
    );
    expect(wrapper.find(DateTime).prop("prepend")).toBe("lorem ipsum");
  });
  it("should pass localized formattedDate to <DateTime />", () => {
    const wrapper = mount(
      <ThemeProvider theme={defaultTokens}>
        <DateFirstPublishedBase story={getDummyStory()} config={config} />
      </ThemeProvider>
    );
    expect(wrapper.find(DateTime).prop("formattedDate")).toBe("February 12, 2020, 4:07 PM");
  });
});
