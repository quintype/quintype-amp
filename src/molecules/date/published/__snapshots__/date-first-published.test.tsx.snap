// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateFirstPublished should match snapshot 1`] = `
<Component
  story={
    Object {
      "access": "subscription",
      "access-level-value": null,
      "alternative": Object {},
      "asana-project-id": null,
      "assignee-id": 708282,
      "assignee-name": "Ravi Kanth",
      "associated-series-collection-ids": Array [],
      "author-id": 708282,
      "author-name": "Ravi Kanth",
      "authors": Array [
        Object {
          "avatar-s3-key": "barandbench/2020-01/82d5c350-9791-4cdb-88d5-f52b2d98a4ab/Bar_and_Bench_Icon_Purple.png",
          "avatar-url": "https://images.assettype.com/barandbench/2020-01/82d5c350-9791-4cdb-88d5-f52b2d98a4ab/Bar_and_Bench_Icon_Purple.png",
          "bio": null,
          "contributor-role": null,
          "id": 708282,
          "name": "Ravi Kanth",
          "slug": "ravi-kanth",
          "twitter-handle": null,
        },
      ],
      "autotags": Array [],
      "breaking-news-linked-story-id": null,
      "bullet-type": "123",
      "canonical-url": null,
      "cards": Array [],
      "comments": null,
      "content-created-at": 1581326633010,
      "content-type": "story",
      "content-updated-at": 1581503851699,
      "contributors": null,
      "created-at": 1581327522143,
      "custom-slug": "Empowering India’s Future Lawyers and Leaders",
      "entities": Object {},
      "external-id": null,
      "first-published-at": 1581503851412,
      "headline": "#Sponsored: Empowering India’s Future Lawyers and Leaders",
      "hero-image-alt-text": null,
      "hero-image-attribution": "Create",
      "hero-image-caption": null,
      "hero-image-metadata": Object {
        "file-name": "Discover_Law_Logo_1.jpg",
        "file-size": 185634,
        "focus-point": Array [
          1275,
          581,
        ],
        "height": 1162,
        "mime-type": "image/jpeg",
        "width": 2550,
      },
      "hero-image-s3-key": "barandbench/2020-02/43fb44c4-2028-4b90-95ad-dc54aad47a28/Discover_Law_Logo_1.jpg",
      "id": "7f3d5bdb-ec52-4047-ac0d-df4036ec974b",
      "is-amp-supported": true,
      "is-live-blog": false,
      "last-published-at": 1590539519368,
      "linked-entities": Array [],
      "linked-stories": Object {},
      "linked-story-ids": Array [
        "a7a06a3f-8d9a-450c-8eaa-387b7c8c6323",
      ],
      "metadata": Object {
        "card-share": Object {
          "shareable": false,
        },
        "sponsored-by": "Empowering India’s Future Lawyers and Leaders",
      },
      "owner-id": 708282,
      "owner-name": "Ravi Kanth",
      "publish-at": null,
      "published-at": 1581503851412,
      "publisher-id": 829,
      "push-notification": null,
      "push-notification-title": null,
      "read-time": 5,
      "sections": Array [
        Object {
          "collection": Object {
            "id": 29911,
            "name": "News",
            "slug": "news",
          },
          "data": null,
          "display-name": "News",
          "domain-slug": null,
          "id": 14017,
          "name": "News",
          "parent-id": null,
          "section-url": "https://www.barandbench.com/news",
          "slug": "news",
        },
      ],
      "seo": Object {
        "claim-reviews": Object {
          "story": null,
        },
        "meta-description": "#Sponsored: New milestones and initiatives mark LSAC’s ongoing efforts to increase access to legal education in India. - Bar & Bench ",
        "meta-keywords": Array [
          "LSAC",
          "Law School",
          "legal Education",
          "Lawyers",
          "Advocates",
          "India",
          "Law",
        ],
        "meta-title": "Empowering India’s Future Lawyers and Leaders",
      },
      "sequence-no": null,
      "slug": "news/empowering-indias-future-lawyers-and-leaders",
      "status": "published",
      "story-content-id": "7f3d5bdb-ec52-4047-ac0d-df4036ec974b",
      "story-template": "text",
      "story-version-id": "1110cdc2-a1d6-4a6c-9707-f4b9b516d49e",
      "storyline-id": null,
      "storyline-title": null,
      "subheadline": "New milestones and initiatives mark LSAC’s ongoing efforts to increase access to legal education in India",
      "summary": null,
      "tags": Array [
        Object {
          "id": 2038546,
          "meta-description": null,
          "meta-title": null,
          "name": "LSAC",
          "slug": "lsac",
          "tag-type": "Tag",
        },
      ],
      "updated-at": 1581503848382,
      "url": "https://www.barandbench.com/news/empowering-indias-future-lawyers-and-leaders",
      "version": 12,
      "votes": Object {},
      "word-count": 1143,
    }
  }
  type="first"
/>
`;

exports[`DateLastPublished should match snapshot 1`] = `
<Component
  story={
    Object {
      "access": "subscription",
      "access-level-value": null,
      "alternative": Object {},
      "asana-project-id": null,
      "assignee-id": 708282,
      "assignee-name": "Ravi Kanth",
      "associated-series-collection-ids": Array [],
      "author-id": 708282,
      "author-name": "Ravi Kanth",
      "authors": Array [
        Object {
          "avatar-s3-key": "barandbench/2020-01/82d5c350-9791-4cdb-88d5-f52b2d98a4ab/Bar_and_Bench_Icon_Purple.png",
          "avatar-url": "https://images.assettype.com/barandbench/2020-01/82d5c350-9791-4cdb-88d5-f52b2d98a4ab/Bar_and_Bench_Icon_Purple.png",
          "bio": null,
          "contributor-role": null,
          "id": 708282,
          "name": "Ravi Kanth",
          "slug": "ravi-kanth",
          "twitter-handle": null,
        },
      ],
      "autotags": Array [],
      "breaking-news-linked-story-id": null,
      "bullet-type": "123",
      "canonical-url": null,
      "cards": Array [],
      "comments": null,
      "content-created-at": 1581326633010,
      "content-type": "story",
      "content-updated-at": 1581503851699,
      "contributors": null,
      "created-at": 1581327522143,
      "custom-slug": "Empowering India’s Future Lawyers and Leaders",
      "entities": Object {},
      "external-id": null,
      "first-published-at": 1581503851412,
      "headline": "#Sponsored: Empowering India’s Future Lawyers and Leaders",
      "hero-image-alt-text": null,
      "hero-image-attribution": "Create",
      "hero-image-caption": null,
      "hero-image-metadata": Object {
        "file-name": "Discover_Law_Logo_1.jpg",
        "file-size": 185634,
        "focus-point": Array [
          1275,
          581,
        ],
        "height": 1162,
        "mime-type": "image/jpeg",
        "width": 2550,
      },
      "hero-image-s3-key": "barandbench/2020-02/43fb44c4-2028-4b90-95ad-dc54aad47a28/Discover_Law_Logo_1.jpg",
      "id": "7f3d5bdb-ec52-4047-ac0d-df4036ec974b",
      "is-amp-supported": true,
      "is-live-blog": false,
      "last-published-at": 1590539519368,
      "linked-entities": Array [],
      "linked-stories": Object {},
      "linked-story-ids": Array [
        "a7a06a3f-8d9a-450c-8eaa-387b7c8c6323",
      ],
      "metadata": Object {
        "card-share": Object {
          "shareable": false,
        },
        "sponsored-by": "Empowering India’s Future Lawyers and Leaders",
      },
      "owner-id": 708282,
      "owner-name": "Ravi Kanth",
      "publish-at": null,
      "published-at": 1581503851412,
      "publisher-id": 829,
      "push-notification": null,
      "push-notification-title": null,
      "read-time": 5,
      "sections": Array [
        Object {
          "collection": Object {
            "id": 29911,
            "name": "News",
            "slug": "news",
          },
          "data": null,
          "display-name": "News",
          "domain-slug": null,
          "id": 14017,
          "name": "News",
          "parent-id": null,
          "section-url": "https://www.barandbench.com/news",
          "slug": "news",
        },
      ],
      "seo": Object {
        "claim-reviews": Object {
          "story": null,
        },
        "meta-description": "#Sponsored: New milestones and initiatives mark LSAC’s ongoing efforts to increase access to legal education in India. - Bar & Bench ",
        "meta-keywords": Array [
          "LSAC",
          "Law School",
          "legal Education",
          "Lawyers",
          "Advocates",
          "India",
          "Law",
        ],
        "meta-title": "Empowering India’s Future Lawyers and Leaders",
      },
      "sequence-no": null,
      "slug": "news/empowering-indias-future-lawyers-and-leaders",
      "status": "published",
      "story-content-id": "7f3d5bdb-ec52-4047-ac0d-df4036ec974b",
      "story-template": "text",
      "story-version-id": "1110cdc2-a1d6-4a6c-9707-f4b9b516d49e",
      "storyline-id": null,
      "storyline-title": null,
      "subheadline": "New milestones and initiatives mark LSAC’s ongoing efforts to increase access to legal education in India",
      "summary": null,
      "tags": Array [
        Object {
          "id": 2038546,
          "meta-description": null,
          "meta-title": null,
          "name": "LSAC",
          "slug": "lsac",
          "tag-type": "Tag",
        },
      ],
      "updated-at": 1581503848382,
      "url": "https://www.barandbench.com/news/empowering-indias-future-lawyers-and-leaders",
      "version": 12,
      "votes": Object {},
      "word-count": 1143,
    }
  }
  type="last"
/>
`;
